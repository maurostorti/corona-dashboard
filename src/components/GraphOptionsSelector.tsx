import { ToggleButtonGroup, ToggleButton } from '@mui/material'
import '../App.css'

type GraphOptionsSelectorProps = {
  graphSelected: string
  onChangeGraphSelected: (value: any) => void
}
export default function GraphOptionsSelector({
  graphSelected,
  onChangeGraphSelected,
}: GraphOptionsSelectorProps) {
  return (
    <ToggleButtonGroup
      value={graphSelected}
      onChange={(event, value) => onChangeGraphSelected(value)}
      orientation='vertical'
      exclusive
      size='small'
    >
      <ToggleButton value='all'> All </ToggleButton>
      <ToggleButton value='confirmed'> Confirmed</ToggleButton>
      <ToggleButton value='recovered'> Recovered</ToggleButton>
      <ToggleButton value='deaths'> Dead </ToggleButton>
    </ToggleButtonGroup>
  )
}
