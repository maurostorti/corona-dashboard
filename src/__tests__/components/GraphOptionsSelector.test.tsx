import { render, screen } from '@testing-library/react'
import GraphOptionsSelector from '../../components/GraphOptionsSelector'

const onChangeGraphSelectedTest = (value: any) => null

test('it renders correctly confirmed', () => {
  const { asFragment } = render(
    <GraphOptionsSelector graphSelected='all' onChangeGraphSelected={onChangeGraphSelectedTest} />,
  )
  expect(asFragment()).toMatchSnapshot()
})

test('it renders buttons', () => {
  render(
    <GraphOptionsSelector graphSelected='all' onChangeGraphSelected={onChangeGraphSelectedTest} />,
  )
  const allButton = screen.getByText('All')
  expect(allButton).toBeInTheDocument()

  const confirmedButton = screen.getByText('Confirmed')
  expect(confirmedButton).toBeInTheDocument()

  const recoveredButton = screen.getByText('Recovered')
  expect(recoveredButton).toBeInTheDocument()

  const deadButton = screen.getByText('Dead')
  expect(deadButton).toBeInTheDocument()
})
