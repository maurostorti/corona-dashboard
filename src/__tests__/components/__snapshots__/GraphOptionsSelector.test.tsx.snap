// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly confirmed 1`] = `
<DocumentFragment>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-vertical css-mjlvw-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-1x4vipr-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="all"
    >
       All 
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-1x4vipr-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="confirmed"
    >
       Confirmed
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-1x4vipr-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="recovered"
    >
       Recovered
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-1x4vipr-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="deaths"
    >
       Dead 
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
